src/app/cli/interpreter.cpp

```diff
@@ -1432,20 +1433,32 @@ Interpreter::Value Interpreter::ProcessBr(const Expression &aExpr)
         nlohmann::json                            baJson;
         char                                      mdnsSendBuffer[kMdnsBufferSize];
 
-        if (mContext.mCommandKeys.size() == 2 && mContext.mCommandKeys[0] == "--timeout")
-        {
+        auto it = std::find(mContext.mCommandKeys.begin(), mContext.mCommandKeys.end(), "--timeout");
+
+        if (it != mContext.mCommandKeys.end()) {
             try
             {
-                scanTimeout = stol(mContext.mCommandKeys[1]);
+              scanTimeout = stol(mContext.mCommandKeys[std::distance(mContext.mCommandKeys.begin(), it) + 1]);
```

Comment:
I think you can use `it[1]`:
```suggestion
              scanTimeout = stol(it[1]);
```